src/Device/Driver/LX_EOS/LXEosDownload.cpp

```diff
@@ -0,0 +1,245 @@
+#include "Device/Error.hpp"
+#include "Device/Port/Port.hpp"
+#include "Device/RecordedFlight.hpp"
+#include "LXEosDevice.hpp"
+#include "Operation/Operation.hpp"
+#include "io/BufferedOutputStream.hxx"
+#include "io/FileOutputStream.hxx"
+#include "system/Path.hpp"
+#include "util/AllocatedArray.hxx"
+#include "util/ByteOrder.hxx"
+#include "util/CRC.hpp"
+#include "util/ScopeExit.hxx"
+
+bool
+LXEosDevice::ReadFlightList(RecordedFlightList& flight_list,
+                            OperationEnvironment& env)
+{
+  port.StopRxThread();
+
+  uint8_t flight_count = 0;
+
+  env.SetProgressRange(1);
+  env.SetProgressPosition(0);
+
+  bool success = GetNumberOfFlights(flight_count, env);
+
+  // Maximum number of flights to be obtained is limited by length of the list
+  if (flight_count > flight_list.size())
+    flight_count = flight_list.size();
+
+  env.SetProgressRange(flight_count + 1);
+  env.SetProgressPosition(1);
+
+  if (success) {
+    for (uint8_t i = 0; i < flight_count; i++) {
+      RecordedFlightInfo flight;
+
+      // Try up to 5 times before giving up
+      for (uint8_t attempt = 0; attempt < 5; attempt++) {
```

Comment:
Are you aware that declaring this `uint8_t` is not faster, but actually adds overhead? It's a fallacy.